pkg/npm/npm.go

```diff
@@ -65,3 +145,207 @@ func registryIsNonEmpty(preConfiguredRegistry string) bool {
+		if err != nil {
+			return nil, fmt.Errorf("failed to read %s to check for existence of %s script: %w", file, script, err)
+		}
+
+		err = json.Unmarshal(packageRaw, &packageJSON)
+		if err != nil {
+			return nil, fmt.Errorf("failed to unmarshal %s to check for existence of %s script: %w", file, script, err)
+		}
+
+		scripts, ok := packageJSON["scripts"].(map[string]interface{})
+		if ok {
+			_, ok := scripts[script].(string)
+			if ok {
+				packagesWithScript = append(packagesWithScript, file)
+				log.Entry().Info("Discovered " + script + " script in " + file)
+			}
+		}
+	}
+	return packagesWithScript, nil
+}
+
+// InstallAllDependencies executes npm or yarn Install for all package.json fileUtils defined in packageJSONFiles
+func (exec *execute) InstallAllDependencies(packageJSONFiles []string) error {
+	for _, packageJSON := range packageJSONFiles {
+		err := exec.install(packageJSON)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// install executes npm or yarn Install for package.json
+func (exec *execute) install(packageJSON string) error {
+	execRunner := exec.utils.getExecRunner()
+
+	oldWorkingDirectory, err := exec.utils.Getwd()
+	if err != nil {
+		return fmt.Errorf("failed to get current working directory before executing npm scripts: %w", err)
+	}
+
+	dir := filepath.Dir(packageJSON)
+	err = exec.utils.Chdir(dir)
+	if err != nil {
+		return fmt.Errorf("failed to change into directory for executing script: %w", err)
+	}
+
+	err = exec.SetNpmRegistries()
+	if err != nil {
+		return err
+	}
+
+	packageLockExists, yarnLockExists, err := exec.checkIfLockFilesExist()
+	if err != nil {
+		return err
+	}
+
+	log.Entry().WithField("WorkingDirectory", dir).Info("Running Install")
+	if packageLockExists {
+		err = execRunner.RunExecutable("npm", "ci")
+		if err != nil {
+			return err
+		}
+	} else if yarnLockExists {
+		err = execRunner.RunExecutable("yarn", "install", "--frozen-lockfile")
+		if err != nil {
+			return err
+		}
+	} else {
+		log.Entry().Warn("No package lock file found. " +
+			"It is recommended to create a `package-lock.json` file by running `npm Install` locally." +
+			" Add this file to your version control. " +
+			"By doing so, the builds of your application become more reliable.")
+		err = execRunner.RunExecutable("npm", "install")
+		if err != nil {
+			return err
+		}
+	}
+
+	err = exec.utils.Chdir(oldWorkingDirectory)
+	if err != nil {
+		return fmt.Errorf("failed to change back into original directory: %w", err)
+	}
+	return nil
+}
+
+// checkIfLockFilesExist checks if yarn/package lock fileUtils exist
+func (exec *execute) checkIfLockFilesExist() (bool, bool, error) {
+	packageLockExists, err := exec.utils.FileExists("package-lock.json")
+	if err != nil {
+		return false, false, err
+	}
+
+	yarnLockExists, err := exec.utils.FileExists("yarn.lock")
+	if err != nil {
+		return false, false, err
+	}
+	return packageLockExists, yarnLockExists, nil
+}
```

Comment:
From what I saw this morning, this doesn't seem to be the latest version of the code. Just wondering whether you are aware.